parser/indexedfile.go

```diff
@@ -55,22 +57,35 @@ func newIndexedFile(filePath string, res *resources.Resources) (*fpt.IndexedFile
 	}
 	toReturn.SetHeader(header)
 
-	broDataFactory := pt.NewBroDataFactory(header.ObjType)
+	var broDataFactory func() pt.BroData
+	if header.ObjType != "" {
+		// TSV log files have the type in a header
+		broDataFactory = pt.NewBroDataFactory(header.ObjType)
+	} else if scanner.Err() == nil && len(scanner.Text()) > 0 && // no error and there is text
+		json.Valid(scanner.Bytes()) {
+		// JSON log files only have the type in the filename
+		broDataFactory = pt.NewBroDataFactory(filepath.Base(toReturn.Path))
+		toReturn.SetJSON()
+	}
 	if broDataFactory == nil {
 		fileHandle.Close()
 		return toReturn, errors.New("Could not map file header to parse type")
 	}
 	toReturn.SetBroDataFactory(broDataFactory)
 
-	fieldMap, err := mapBroHeaderToParserType(header, broDataFactory, res.Log)
-	if err != nil {
-		fileHandle.Close()
-		return toReturn, err
+	var fieldMap fpt.BroHeaderIndexMap
+	// there is no need for the fieldMap with JSON
+	if !toReturn.IsJSON() {
+		fieldMap, err = mapBroHeaderToParserType(header, broDataFactory, res.Log)
+		if err != nil {
+			fileHandle.Close()
+			return toReturn, err
+		}
+		toReturn.SetFieldMap(fieldMap)
```

Comment:
Since json files don't have a header this block is unnecessary and causes an error.